sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -214,7 +215,12 @@ inline bool operator==(DateTime const& dt, std::chrono::system_clock::time_point
 
 inline bool operator<(DateTime const& dt, std::chrono::system_clock::time_point const& tp)
 {
-  return dt < DateTime(tp);
+  if (dt < DateTime(tp)) // LCOV_EXCL_LINE
+  {
+    return true;
+  }
+
+  return false;
```

Comment:
It isn't great that we have to contort the code-style to be a bit more confusing in order to satisfy code-coverage.

The immediate question for code snippet like this would be, why is it not written the way it was previously.

Why is `dt < DateTime(tp);` not being able to get covered?

Reply:
It is covered, but gcovr gives false positive. Doesn't do well for inline functions, I thik.

That extra dance with return true/false is to show when you look at the lines, that both returns are green lines, i.e. covered.